src/pybind/mgr/cephadm/tests/test_services.py

```diff
@@ -899,6 +902,114 @@ def test_snmp_v3priv_deployment(self, _run_cephadm, cephadm_module: CephadmOrche
 
 class TestIngressService:
 
+    @patch("cephadm.inventory.Inventory.get_addr")
+    @patch("cephadm.utils.resolve_ip")
+    @patch("cephadm.inventory.HostCache.get_daemons_by_service")
+    @patch("cephadm.serve.CephadmServe._run_cephadm")
+    def test_ingress_config_nfs_multiple_nfs_same_rank(self, _run_cephadm, _get_daemons_by_service, _resolve_ip, _get_addr, cephadm_module: CephadmOrchestrator):
+        _run_cephadm.side_effect = async_side_effect(('{}', '', 0))
+
+        def fake_resolve_ip(hostname: str) -> str:
+            if hostname == 'host1':
+                return '192.168.122.111'
+            elif hostname == 'host2':
+                return '192.168.122.222'
+            else:
+                return 'xxx.xxx.xxx.xxx'
+        _resolve_ip.side_effect = fake_resolve_ip
+
+        def fake_get_addr(hostname: str) -> str:
+            return hostname
+        _get_addr.side_effect = fake_get_addr
+
+        nfs_service = NFSServiceSpec(service_id="foo", placement=PlacementSpec(count=1, hosts=['host1', 'host2']),
+                                     port=12049)
+
+        ispec = IngressSpec(service_type='ingress',
+                            service_id='nfs.foo',
+                            backend_service='nfs.foo',
+                            frontend_port=2049,
+                            monitor_port=9049,
+                            virtual_ip='192.168.122.100/24',
+                            monitor_user='admin',
+                            monitor_password='12345',
+                            keepalived_password='12345')
+
+        cephadm_module.spec_store._specs = {
+            'nfs.foo': nfs_service,
+            'ingress.nfs.foo': ispec
+        }
+        cephadm_module.spec_store.spec_created = {
+            'nfs.foo': datetime_now(),
+            'ingress.nfs.foo': datetime_now()
+        }
+
+        haproxy_expected_conf = {
+            'files':
```

Comment:
I think it would be good to add some comment explaining that `host1` is used in the following conf (because is the host with the higher rank).

Reply:
added comment